tests/approvals/tests.py

```diff
@@ -1984,3 +1985,77 @@ def test_approval_can_be_unsuspended(reason, outcome):
         reason=reason,
     )
     assert ja.approval.can_be_unsuspended is outcome
+
+
+def test_get_user_last_accepted_siae_job_application():
+    # Set 2 job applications with:
+    # - origin set to PE_APPROVAL (the simplest method to test created_at ordering)
+    # - different creation date
+    # `last_accepted_job_application` is the one with the greater `created_at`
+    now = timezone.now()
+    job_application_1 = JobApplicationFactory(
+        state=JobApplicationState.ACCEPTED,
+        to_company__subject_to_eligibility=True,
+        origin=Origin.PE_APPROVAL,
+        created_at=now + relativedelta(days=1),
+    )
+
+    user = job_application_1.job_seeker
+
+    job_application_2 = JobApplicationFactory(
+        job_seeker=user,
+        state=JobApplicationState.ACCEPTED,
+        to_company__subject_to_eligibility=True,
+        origin=Origin.PE_APPROVAL,
+        created_at=now,
+    )
+
+    assert job_application_1 == get_user_last_accepted_siae_job_application(user)
+    assert job_application_2 != get_user_last_accepted_siae_job_application(user)
+
+
+def test_get_user_last_accepted_siae_job_application_full_ordering():
+    # Set 2 job applications with:
+    # - origin set to PE_APPROVAL (the simplest method to test created_at ordering)
+    # - same creation date
+    # - different hiring date
+    # `last_accepted_job_application` is the one with the greater `hiring_start_at`
+    now = timezone.now()
+    job_application_1 = JobApplicationFactory(
+        state=JobApplicationState.ACCEPTED,
+        to_company__subject_to_eligibility=True,
+        origin=Origin.PE_APPROVAL,
+        created_at=now,
+        hiring_start_at=now + relativedelta(days=1),
+    )
+
+    user = job_application_1.job_seeker
+
+    job_application_2 = JobApplicationFactory(
+        job_seeker=user,
+        state=JobApplicationState.ACCEPTED,
+        to_company__subject_to_eligibility=True,
+        origin=Origin.PE_APPROVAL,
+        created_at=now,
+        hiring_start_at=now,
+    )
+
+    assert job_application_1 == get_user_last_accepted_siae_job_application(user)
+    assert job_application_2 != get_user_last_accepted_siae_job_application(user)
+
+
+def test_last_hire_was_made_by_siae():
+    siae_job_application = JobApplicationSentByJobSeekerFactory(
+        state=JobApplicationState.ACCEPTED,
+        to_company__subject_to_eligibility=True,
+    )
+    user = siae_job_application.job_seeker
+    new_geiq_job_application = JobApplicationSentByJobSeekerFactory(
```

Comment:
Vu qu'on cible pas spécifiquement les GEIQ et qu'a l'avenir on pourrais avoir d'autre type d'entreprise non-SIAE
```suggestion
    non_siae_job_application = JobApplicationSentByJobSeekerFactory(
```